server/py/services/api/launcher.py

```diff
@@ -697,20 +697,31 @@ def _validate_retry(runtime_kind: str, retry: Optional["mlrun.model.Retry"]):
                 )
 
     # TODO In ML-11600, implement token name resolution and validation + tests
-    def _enrich_and_validate_auth_token_name(self, run: mlrun.run.RunObject):
-        auth = run.spec.auth or {}
+    def enrich_and_validate_auth_token_name_on_object(
+        self, object: Union[mlrun.run.RunObject, mlrun.runtimes.RemoteRuntime]
+    ):
+        # Ensure the auth dictionary exists
+        if object.spec.auth is None:
+            object.spec.auth = {}
 
-        if auth.get("token_name"):
-            self._validate_token_name(
-                auth["token_name"],
-                explicit=True,
-            )
-            return
+        # Get the provided token name, if any
+        provided_token_name = object.spec.auth.get("token_name")
 
-        run.spec.auth["token_name"] = "default"
+        # Resolve token name
+        if provided_token_name:
+            token_name = provided_token_name
+            self._validate_token_name(token_name, raise_error_on_failure=True)
+        else:
+            token_name = "default"
+            self._validate_token_name(token_name, raise_error_on_failure=False)
```

Comment:
is default token name a constant? I think it should be
also, comment on why you raise an exception?



```suggestion
        # explain me
        raise_error_on_failure = bool(provided_token_name)
        token_name = provided_token_name or "default"
        self._validate_token_name(token_name, raise_error_on_failure=False)
```

Reply:
@quaark I can use `mlconf.auth_with_oauth_token.token_name`, however if user sets this env var prior to importing mlrun, then runs/deploy a function, the token resolution logic will use that instead. 
I assume we don't want that behavior right?

Reply:
Ok so we'll make it a constant not a config